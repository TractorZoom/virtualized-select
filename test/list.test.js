import React from 'react'
import List from '../src/list'
import renderer from 'react-test-renderer'

test('List when empty', () => {
    const component = renderer.create(<List children={[]} getValue={() => []} options={[]} />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('List when less than 6 options', () => {
    const childrenAndOptions = ['first child', 'second child']
    const component = renderer.create(
        <List children={childrenAndOptions} getValue={() => []} options={childrenAndOptions} />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('List when value set and option found', () => {
    const childrenAndOptions = ['first child', 'second child']
    const component = renderer.create(
        <List children={childrenAndOptions} getValue={() => ['first option']} options={childrenAndOptions} />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('List when value set and option not found', () => {
    const childrenAndOptions = [
        'first child',
        'second child',
        'third child',
        'fourth child',
        'fifth child',
        'sixth child',
        'seventh child',
    ]
    const component = renderer.create(
        <List children={childrenAndOptions} getValue={() => ['some other option']} options={childrenAndOptions} />
    )
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})
