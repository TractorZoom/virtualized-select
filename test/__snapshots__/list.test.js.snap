// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List when empty 1`] = `
<div
  className="auto-sized-menu"
>
  <div
    style={
      Object {
        "height": 0,
        "overflow": "visible",
        "width": 0,
      }
    }
  />
</div>
`;

exports[`List when less than 6 options 1`] = `
<div
  className="auto-sized-menu"
>
  <div
    style={
      Object {
        "height": 0,
        "overflow": "visible",
        "width": 0,
      }
    }
  />
</div>
`;

exports[`List when value set and option found 1`] = `
<div
  className="auto-sized-menu"
>
  <div
    style={
      Object {
        "height": 0,
        "overflow": "visible",
        "width": 0,
      }
    }
  />
</div>
`;

exports[`List when value set and option not found 1`] = `
<div
  className="auto-sized-menu"
>
  <div
    style={
      Object {
        "height": 0,
        "overflow": "visible",
        "width": 0,
      }
    }
  />
</div>
`;
